### PR TITLE
ts strict part 5: misc 1

### PR DESCRIPTION
**Pull request by chrisnojima**
Please see https://github.com/keybase/client/pull/18383 for an overview

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/desktop/remote/component-loader.desktop.tsx [hunk 1]
```diff
@@ -97,7 +97,7 @@ export default function(options: {
       <RemoteComponentLoader
         name={options.name}
         params={options.params || ''}
-        style={options.style}
+        style={options.style || null}
```
Comment (MarcoPolo) [inline shared/desktop/remote/component-loader.desktop.tsx]: why null instead of undefined here?

File: shared/router-v2/router.native.tsx [hunk 2]
```diff
@@ -254,12 +255,15 @@ class RNApp extends React.PureComponent<any, any> {
...
 
-  getNavState = () =>
-    // Auto generated from flowToTs. Please clean me!
-    (this._nav === null || this._nav === undefined ? undefined : this._nav.state) === null ||
-    (this._nav === null || this._nav === undefined ? undefined : this._nav.state) === undefined
-      ? undefined
-      : (this._nav === null || this._nav === undefined ? undefined : this._nav.state).nav
+  getNavState = () => {
+    const n = this._nav
+    if (n) {
```
Comment (MarcoPolo) [inline shared/router-v2/router.native.tsx]: 🐑 return `n && n.state && n.state.nav || null`

File: shared/settings/chat/index.tsx [hunk 3]
```diff
@@ -29,10 +29,12 @@ class Chat extends React.Component<Props, State> {
     )
   }
   _getUnfurlMode() {
-    return this.state.unfurlSelected !== undefined
-      ? this.state.unfurlSelected
-      : this.props.unfurlMode !== undefined
-      ? this.props.unfurlMode
+    const unfurlSelected = this.state.unfurlSelected
+    const unfurlMode = this.props.unfurlMode
+    return unfurlSelected !== undefined
```
Comment (MarcoPolo) [inline shared/settings/chat/index.tsx]: 🐑 :sheep: Maybe worth expanding this? It's a little confusing.

File: shared/team-building/container.tsx [hunk 4]
```diff
@@ -376,15 +376,19 @@ const mergeProps = (
     onChangeService: ownProps.onChangeService,
     onChangeText,
     onClosePopup: dispatchProps._onCancelTeamBuilding,
-    onDownArrowKeyDown: ownProps.showRolePicker
-      ? rolePickerArrowKeyFns.downArrow
-      : deriveOnDownArrowKeyDown((userResultsToShow || []).length - 1, ownProps.incHighlightIndex),
+    onDownArrowKeyDown:
+      ownProps.showRolePicker && rolePickerArrowKeyFns
```
Comment (MarcoPolo) [inline shared/team-building/container.tsx]: Nice catch, did TS throw an error here?

File: shared/desktop/remote/component-loader.desktop.tsx (hunk 1 above)
Comment (chrisnojima) [reply shared/desktop/remote/component-loader.desktop.tsx]: can be, i didn't update all instances of the null vs undefined thing but we should

File: shared/team-building/container.tsx (hunk 4 above)
Comment (chrisnojima) [reply shared/team-building/container.tsx]: yup

**Pull request merged**